src/Ring.cc

```diff
@@ -190,7 +222,7 @@ void Ring::buildTopDown() {
 
   if (numModules.state()) numMods = numModules();
   else numModules(numMods);
-  buildModules(emod, numMods, smallDelta());
+  buildModules(emod, numMods, smallDelta(),phiShift());
```

Comment:
Same, would be nice to be more specific about what phiShift represents.

Alright, sorry for the long review. These are only suggestions, hope it helps!
Cheers